astore/server/auth/auth_test.go

```diff
@@ -0,0 +1,175 @@
+EAAAGBAPD8+8CoZnxvM/HtpfsKIbVBRhIF4SUYASyb+ojw9Fw8yTHq3OF9hPB/7sQ+eCcf
+IK4p1Hz2/UM0zzDODkIoIHh/rSuN/JhMC03viwURehu02R8wjuZYjaz8/SLc1vB+cdbDg4
+PQlSRbuu1nAC36k7tkZXrwgad6h9Q67C9X9Eisk052pvaL1bgTTsKCjZBwN2j9LU1rtgT5
+NnQQQ4W238A7IPiNP1BtA/KtO1YkInZs01OZFplKWfvnF5wwK+CqcROnuN041aIwrfy2nS
+diaygt42/CmG7O4zUfO8CXsQgq9bnFV58qstnoZOKKyY1K7RseeQ6RrXRvlA3/BF6tQVn7
+ouAz+fJ9c/0CcPTDj0gcRIehBijdVq2lZjSKqjuYYC29zXQ6INoJb9pm8VlRi30F+15zE0
+rbWd+b9AUH8+BVkbXSXMr/lpflXaMq2aCVAXTSMV6L/rhfFs+MxPzN4j/Setc9qOCA9MAS
+jr6WNXIjGvK8zQPnUaABX9pag9Q00QAAAAMBAAEAAAGBAL2IuxgTWkeTzm8AUgLXPRupcs
+rKBQF/l6zWIH2DxSymQjcYWRCf/+aHN+rwlt9uA+32yEBgoWAyMKJZ7azqkl8zS6dtzLSb
+Wmi5dcVOsZMI8ZsuPbW8//CGKTE6L3KGgFJBAzaw3hvyaVo+IE4JPhesJoRClDZ8kEfC7+
+9sZZyi3lhfyYEvCa/0v4UL2Ps4xtu0A+VYSZgvyTwPbovEAMbXul7B+IHwu6IpzPk7Aj/R
+54Nga/20FIGih1c4K8pPQYk3DCVDg/VUkFwiugnzokwDQGPkIcSMuvPoyyZrctKSBjS9T0
+krul1+9HqdsK0IU449n5Z1FciexYq88l5lBmih/H3HrIaAIj8nnkRVMX7n4kr4w+98aAQt
+c4ZpwA3q3EfKFMBbSu9mye85D5qdtRxKIdSCNqgDONcOyjs+0euumal7YoB3UMoVExO3Lr
+hoC9yCVdFAwlfO8IwCDTpwu3mbGiAyfMlcs5Mi3QQLN8AnggJAfU/8QOMQ4b+x8KMF4QAA
+AMEAxW+kPIj59Cw360MvW2GcvXWYYOTnK+pUAfDWHbj6s7aVBxaQwZ5r0efzYaleucLB42
+y1IKiOK8P/QULel9+5qqnaCVQRHn6Ob0DBphGYuWnEw4rr1itt5JTe0Q7Ceb/nVKEvk1Rh
+dhxF2AH4VaGqxvFYUlOWL98+vUCwR6w30FNiyb1uBgtYzFk3Vmb+RzwQGo40Xh1lbYnhz6
+fdlHxwwP656kS71huk5pDTGpikfg5i+NTqmwKcezXCNHo003GzAAAAwQD+2TJH8nndcWqG
+PQTdahn89vtVbOhi1N4+wqNV1GfwTe61t88T4yFi6xA67OabmF3hJdmz9oA+D4Opmr/kXV
+a8VfMQEQ6oB3a7IQP+1y4FUjWOQKYuppO0WID1/WJ8hJ9FnWNaG1O/wMxvkyoYCGo7Sftp
+MCyKIiF8n3tpKuIpYVWmpzon8GJiGCNzDGrtOgpfepgMJ7Jrvk3KJTprwJpZ+sHUZt6QOV
+MDdiNlQrFsOLMqnJhLf0yIlIRUkhZBLKMAAADBAPITwPgh4KGzMjcV377WICPWGFfRqpmX
+uOj+J5RAR42wAYnvjf66k+wTiBnq/fL2xzepKsvw+pqZXkLskPKB9MyR6fu+GtxCDfLhy8
+imdebXFcEZasiV7Lr4yDhTy03c/ZDhiMOKsyTDw7C+FwVNV8ziHPhM+lSFZ3WA70gFiBEl
+PReG8EaryR9nIKK8y9Y/QAK3Yjvo8uWsYrGiYTjQnmy+UJOGG4JBr+Htl9r5V7Q6asDDRc
+NxwAnyaoTZyirb+wAAABFjY29udGF2YWxsaUBub3JhZA==
+-----END OPENSSH PRIVATE KEY-----`
+
+func TestCAAuthBroken(t *testing.T) {
+	rng := rand.New(srand.Source)
+
+	whacky := []byte("I don't need no certificate")
+	server, err := New(rng, WithAuthURL("static-prefix"), WithCA(whacky))
+	assert.NotNil(t, err, err)
+	assert.Nil(t, server)
+}
+
+func TestCAAuthRSA(t *testing.T) {
+	rng := rand.New(srand.Source)
+
+	server, err := New(rng, WithAuthURL("static-prefix"), WithCA([]byte(rsaTestKey)))
+	assert.Nil(t, err, err)
+	assert.NotNil(t, server)
+
+	tresp := Authenticate(t, rng, server)
+	assert.Less(t, 128, len(tresp.Key), "%v", tresp.Key)
+	assert.Less(t, 128, len(tresp.Cert), "%v", tresp.Cert)
+	assert.Less(t, 128, len(tresp.Capublickey), "%v", tresp.Capublickey)
+}
+
+// This is actually the real private key that gives you access to
+// all of the NASA infrastructure, go use it.
+//
+// (joking of course, if your security scanner goes crazy on this,
+// go read the comment a few paragraphs above).
+var edTestKey = `-----BEGIN OPENSSH PRIVATE KEY-----
```

Comment:
Ideally I would use go's crypto lib for this because there's no way for testers to actually verify this is an ecdsa key because its pem encoded. 

I'm fine with embedding as is, but I don't know how scalable it is.

```
func newECDSA() string {
    privateKey, _ := ecdsa.GenerateKey(elliptic.P384(), rand.Reader)
    x509Encoded, _ := x509.MarshalECPrivateKey(privateKey)
    return pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: x509Encoded})
}
```
and something similar for rsa.

Reply:
As mentioned - good comment. I will send a separate PR to add a test for the ECDSA key generated via go, my intent here was to verify with a key literally generated by openssh.